aggregative_results/aggregators/energy_model_aggregator.py

```
@@ -0,0 +1,155 @@
+            network_bytes_kb_sum_received_system=system_data.packets_received,
+            network_packets_sum_received_system=system_data.packets_received,
+            disk_read_time_system_ms_sum=system_data.disk_read_time,
+            disk_write_time_system_ms_sum=system_data.disk_write_time
+        )
+        return EnergyModelFeatures(
+            timestamp=iteration_metadata.timestamp,
+            process_features=process_features,
+            system_features=system_features
+        )
+
+    def process_sample(self, sample: EnergyModelFeatures) -> Union[EnergyModelResult, EmptyAggregationResults]:
+        try:
+            if self.__previous_sample is None:
+                return EmptyAggregationResults()
+
+            sample_as_dict = self.__convert_sample_to_dict(sample)
+            duration = sample.timestamp - self.__previous_sample.timestamp
+            sample_as_dict[DURATION_COLUMN] = duration
+            sample_as_df = pd.DataFrame([sample_as_dict])
+            energy_prediction = self.__model.predict(sample_as_df)
+
+            energy_per_resource = self.__calculate_energy_per_resource(sample, energy_prediction)
+            return EnergyModelResult(energy_mwh=energy_prediction,
+                                     cpu_energy_consumption=energy_per_resource.cpu_energy_consumption,
+                                     ram_energy_consumption=energy_per_resource.ram_energy_consumption,
+                                     disk_io_read_energy_consumption=energy_per_resource.disk_io_read_energy_consumption,
+                                     disk_io_write_energy_consumption=energy_per_resource.disk_io_write_energy_consumption,
+                                     network_io_received_energy_consumption=energy_per_resource.network_io_received_energy_consumption,
+                                     network_io_sent_energy_consumption=energy_per_resource.network_io_sent_energy_consumption)
+        except Exception as e:
```

Why?
Let it crash or at least print a warning.

---

i printed something. I dont really care if you rather to let it crash (however i dont think its the right thing)